### PR TITLE
Update cli project to build cli after repo refactoring

### PR DESCRIPTION
**Pull request by dseefeld**
* Update RootRepo to build through cli for everything but Windows
* Remove build args and flags that no longer apply to cli
* Update RepositoryReferences to only include ones used by cli

### CODE REVIEW

File: dir.props [hunk 1]
```diff
@@ -89,17 +89,10 @@
...
-        When all repos build, this can be changed back
-        to cli.
     <RootRepo>cli</RootRepo>
-    -->
-
-    <!-- Temporarily update RootRepo to core-setup repo project
+    <!-- Temporarily update RootRepo for Windows to core-setup repo project
     until all repos build clean out of master -->
-    <RootRepo>core-setup</RootRepo>
+    <RootRepo Condition="'$(OS))' == 'Windows_NT'">core-setup</RootRepo>
```
Comment (dagood) [inline dir.props]: Typo: extra `)` in `$(OS))`

File: dir.props [hunk 2]
```diff
@@ -89,17 +89,10 @@
...
-    <!--
-        Temporarily update RootRepo to known-good
-        repo project until all repos are building clean.
-        When all repos build, this can be changed back
-        to cli.
     <RootRepo>cli</RootRepo>
-    -->
-
-    <!-- Temporarily update RootRepo to core-setup repo project
+    <!-- Temporarily update RootRepo for Windows to core-setup repo project
```
Comment (dagood) [inline dir.props]: Maybe the comment should be split into one each for Windows vs. arm? (Also justifies having two props setting the same value IMO.)

File: repos/cli.proj [hunk 3]
```diff
@@ -118,37 +91,5 @@
                       Overwrite="true" />
   </Target>
```
Comment (dagood) [inline repos/cli.proj]: Do you mind adding a link to https://github.com/dotnet/cli/issues/9457 tracking adding a feature that lets us remove this target while you're cleaning up?

Comment (dagood) [reply repos/cli.proj]: Or https://github.com/dotnet/source-build/issues/561 actually.

File: dir.props (hunk 2 above)
Comment (dseefeld) [reply dir.props]: Added an additional comment for  arm.

File: dir.props (hunk 1 above)
Comment (dseefeld) [reply dir.props]: Fixed.  Thanks.

File: repos/cli.proj (hunk 3 above)
Comment (dseefeld) [reply repos/cli.proj]: I added a comment with this link.

**Pull request merged**